test/src/test/java/jenkins/model/JenkinsManagePermissionTest.java

```diff
@@ -0,0 +1,214 @@
+package jenkins.model;
+
+import java.net.HttpURLConnection;
+import com.gargoylesoftware.htmlunit.WebResponse;
+import com.gargoylesoftware.htmlunit.html.HtmlForm;
+import com.gargoylesoftware.htmlunit.html.HtmlPage;
+import org.hamcrest.BaseMatcher;
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.Issue;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.jvnet.hudson.test.MockAuthorizationStrategy;
+import org.jvnet.hudson.test.recipes.WithPlugin;
+
+import hudson.PluginWrapper;
+import hudson.cli.CLICommandInvoker;
+import hudson.cli.DisablePluginCommand;
+import hudson.model.Descriptor;
+import hudson.model.MyView;
+import hudson.model.View;
+import hudson.model.labels.LabelAtom;
+import hudson.tasks.Shell;
+
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.not;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+
+import static hudson.cli.CLICommandInvoker.Matcher.failedWith;
+
+/**
+ * As Jenkins.MANAGE can be enabled on startup with jenkins.security.ManagePermission property, we need a test class
+ * with this property activated.
+ */
+public class JenkinsManagePermissionTest {
+
+    @Rule
+    public JenkinsRule j = new JenkinsRule();
+
+    @BeforeClass //TODO: remove once Jenkins.MANAGE is no longer an experimental feature
+    public static void enableManagePermission() {
+        System.setProperty("jenkins.security.ManagePermission", "true");
+    }
+
+    @AfterClass //TODO: remove once Jenkins.MANAGE is no longer an experimental feature
+    public static void disableManagePermission() {
+        System.clearProperty("jenkins.security.ManagePermission");
+    }
+
+
+    // -----------------------------
+    //Moved from DisablePluginCommandTest
```

Comment:
This comment is confusing. As best I can tell, these tests are inspired by tests in `DisablePluginCommandTest` but nothing was moved from there or even directly copied. I'm not sure there's even any value in referencing at all here.

There are other similar confusing comments in this same test file referring to other moves.

I suppose the private helper methods are **copied** from the other files. That copying could be noted here, but I'm not sure that's useful in these tests.

Reply:
These tests logically belong in the other tests, and are only here because of the need to set the system property. That's my interpretation at least.